pkg/docker/util.js

```diff
@@ -595,6 +595,72 @@
         return deferred.promise();
     };
 
+    util.delete_image_confirm = function confirm(client, image) {
+        var deferred = $.Deferred();
+
+        $('#delete-image-confirmation-dialog-title').text(
+            cockpit.format(_("Please confirm deletion of $0"), image.name));
+
+        $('#delete-image-confirmation-dialog-containers>table>tbody').empty();
+        $('#delete-image-confirmation-dialog-body div:nth-child(2)').empty();
+        client.containers_for_image(image.image_id).then(function(containers) {
+            var running_containers = false;
+            $('#delete-image-confirmation-dialog-body div:first-child()').text(
+                _("Deleting an image will delete it, but you can probably download " +
+                  "it again if you need it later.  Unless this image has never been " +
+                  "pushed to a repository, that is, in which case you probably can't " +
+                  "download it again."));
+
+            $(containers).each(function(index, value) {
+                var container = client.containers[value.Id];
+                var $row = $('<tr />')
+                    .append($('<td />').text(container.Name.startsWith('/') ?
+                                             container.Name.substring(1) : container.Name))
+                    .append($('<td />').text(util.render_container_status(container.State)));
+                $('#delete-image-confirmation-dialog-containers>table>tbody').append($row);
+                running_containers |= container.State.Running;
+            });
+
+            if (containers.length > 0) {
+                $('#delete-image-confirmation-dialog-body div:nth-child(2)').text(
+                    _("The following containers depend on this image and will be deleted."));
+                $('#delete-image-confirmation-dialog-containers').show();
+                $('#delete-image-confirmation-dialog-confirm').text(_("Force delete"));
+            } else {
+                $('#delete-image-confirmation-dialog-containers').hide();
+                $('#delete-image-confirmation-dialog-confirm').text(_("Delete"));
+            }
+
+            if (running_containers) {
```

Comment:
We have two choices here:
1. Stop people from bringing up this dialog at all if the image has running containers by disabling the delete button that brings it up and place the message about running containers there. As it works right now, you can only cancel the action, then go and find the place to stop all the containers that the image depend on and get back in. This doesn't feel straightforward to me.
2. Allow to stop all the containers that depend the image and delete the image in one go, naming the button "Stop and Delete" instead.

I like alternative 2 the best myself, but 1 could work as well.